lib/jsdom/living/helpers/form-controls.js

```diff
@@ -214,3 +215,15 @@ exports.sanitizeValueByType = (input, val) => {
 
   return val;
 };
+
+exports.associatedForm = formControl => {
```

Comment:
Yes - as far as I can tell from reading the spec, It doesn't seem necessary to actually follow that imperative implementation for initializing/assigning the form owner based on certain events rather than just calculating the form owner on demand. I'll add a comment to this effect. While it's possible that there is some nuance I don't understand I think this implementation covers the vast majority of use cases.